docs/guides/latest/addons/overview.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Addons | Stash
+menu:
+  product_stash_{{ .version }}:
+    identifier: addons-overview
+    name: Overview
+    parent: addons
+    weight: 10
+product_name: stash
+menu_name: product_stash_{{ .version }}
+section_menu_id: guides
+---
+
+# Stash Addons
+
+Stash 0.9.0+ supports extending its functionality through addons. This guide will give you an overview of what is an addon, how addons work and a list of Stash addons.
+
+## What is an Addon
+
+Stash 0.9.0+ uses two different models for backup/restore based on target types. One of them is **sidecar model** where Stash injects a `sidecar`/`init-container` into the targeted workload for backup/restore. Another is **job model** where backup or restore is done via an external job.
+
+The job model is further divided into two categories. In the first category, the targeted resource is well known to Stash (example, a PVC). Hence, the Stash operator itself can create the required job to backup/restore the target. In the second category, Stash follows `Function-Task` model where the targeted resource is not known to Stash. In this case, the user creates some [Function](/docs/concepts/crds/function.md) which resembles a step of backup/restore process and a [Task](/docs/concepts/crds/task.md) which specifies the order of execution of these steps. Stash uses these `Function` and `Task` to generate the required job definition to backup/restore the target.
+
+The `Function-Task` model enables Stash to backup/restore the resources that the operator itself is not aware of. Users can extend Stash by creating respective `Function` and `Task` to backup their desired resources.
+
+A Stash addon is a collection of [Functions](/docs/concepts/crds/function.md) and a [Task](/docs/concepts/crds/task.md) to backup & restore a specific resource.
```

Comment:
Should we also mention that they need to write the docker images?

Reply:
done